src/platform/common/crypto.ts

```diff
@@ -6,33 +6,23 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
 import { injectable } from 'inversify';
-import { traceError } from '../logging';
-import { ICryptoUtils, IHashFormat } from './types';
+import { ICryptoUtils } from './types';
 import * as hashjs from 'hash.js';
 
 /**
  * Provides hashing functions. These hashing functions should only be used for non sensitive data. For sensitive data, use msrCrypto instead.
  */
 @injectable()
 export class CryptoUtils implements ICryptoUtils {
-    public createHash<E extends keyof IHashFormat>(
-        data: string,
-        hashFormat: E,
-        algorithm: 'SHA512' | 'SHA256' = 'SHA256'
-    ): IHashFormat[E] {
-        let hash: string;
-        if (algorithm === 'SHA256') {
-            hash = hashjs.sha256().update(data).digest('hex');
-        } else {
-            hash = hashjs.sha512().update(data).digest('hex');
-        }
-        if (hashFormat === 'number') {
-            const result = parseInt(hash, 16);
-            if (isNaN(result)) {
-                traceError(`Number hash for data '${data}' is NaN`);
-            }
-            return result as any;
-        }
-        return hash as any;
+    public createHash(data: string, algorithm: 'SHA512' | 'SHA256' = 'SHA256'): string {
+        return computeHash(data, algorithm);
+    }
+}
+
+export function computeHash(data: string, algorithm: 'SHA512' | 'SHA256') {
+    if (algorithm === 'SHA256') {
```

Comment:
This is one of the only two places where we generate hashes,
plan is to merge the others with this and have this code to return the async version of hashing, which would be a bigger change (hence a separate PR).
Big chage because the new hashing code will be async.